### PR TITLE
Feature/track name

### PR DESCRIPTION
**Pull request by DominikGuzei**
Adds the possibility to name your tracks and see these names in the tracks screen. The track title can be entered in the publication step in the builder. If no title is provided, a random one is taken from an array of pre-defined default titles.

existing tracks like in the online version will have an empty title, and thus it will not be visible on the tracks screen. There two possibilities: delete all existing ones or run a seed on production that adds random names to all "untitled" tracks.

### CODE REVIEW

**Comment by stravid** [comment]
Nice! But your tests fail :o
http://travis-ci.org/#!/stravid/slotcars/builds/1400648

**Comment by DominikGuzei** [comment]
oops.. thanks -> these errors didn't show up at first on my machine :-) will fix that

File: app/assets/javascripts/slotcars/build/templates/publication_view_template.js.hjs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 <div id="publication-dialog">
+  <input type="text" id="track-title-input" placeholder="Enter Track Title.." />
```
Comment (tombu) [inline app/assets/javascripts/slotcars/build/templates/publication_view_template.js.hjs]: please use '...' (3 dots) or the HTML entity '&hellip;' in the placeholder - or no dot

File: app/assets/javascripts/slotcars/shared/models/track.js.coffee [hunk 2]
```diff
@@ -64,10 +65,26 @@ Shared.Track = DS.Model.extend Shared.Rasterizable,
...
+
 Shared.Track.reopenClass
 
   MINIMUM_TRACK_LENGTH: 400
 
+  # one of these is used if no title provided on publishing
+  RANDOM_TRACK_TITLES: [
+    'Just Another Awesome One'
+  ]
+
```
Comment (tombu) [inline app/assets/javascripts/slotcars/shared/models/track.js.coffee:85]: one-liner?

File: spec/javascripts/unit/slotcars/build/publisher_spec.js.coffee [hunk 3]
```diff
@@ -75,3 +75,23 @@ describe 'Build.Publisher', ->
         publicationCallback() # gets normally called by track.save
 
         (expect Shared.routeManager.set).toHaveBeenCalledWith 'location', "play/#{@trackId}"
+
+      it 'should set the title on the track before saving', ->
+        @publisher.publish()
+
+        (expect @publisher.setTitleOnTrackToPublish.calledBefore @trackMock.save).toBe true
```
Comment (tombu) [inline spec/javascripts/unit/slotcars/build/publisher_spec.js.coffee]: I prefer to use `toHaveBeenCalledBefore()`...

File: spec/javascripts/unit/slotcars/shared/models/track_spec.js.coffee [hunk 4]
```diff
@@ -302,3 +284,33 @@ describe 'Shared.Track', ->
         @requests[0].respond 200, { "Content-Type": "application/json" }, JSON.stringify @responseJSON
 
         (expect record.get 'id').toEqual 42
+
+
+  describe 'setting title of track', ->
+
+    describe 'valid title provided', ->
```
Comment (tombu) [inline spec/javascripts/unit/slotcars/shared/models/track_spec.js.coffee]: If we describe a special case I like starting it with the word `'when ... '`\- it´s more intuitive to read, e.g `when ... it should ...` (also in l.301)

**Comment by tombu** [comment]
Apart from some minor issues I like it!

**Pull request merged**